### PR TITLE
feat: add serif implementation

### PR DESCRIPTION
**Pull request by jceb**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "2.1.0",
   "description": "Fantasy Land -compliant Either type",
   "license": "MIT",
+  "type": "module",
```
Comment (jceb) [inline package.json:6]: For this change to work, we'd have convert the existing implementation .. are there other options?

File: index.serif [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+#| > S.concat (Left ("abc")) (Right ([1, 2, 3]))
+#| Right ([1, 2, 3])
+#|
+#| > S.concat (Right ([1, 2, 3])) (Left ("abc"))
+#| Right ([1, 2, 3])
+#| ```
+let LeftXprototypeXconcat = (this, that) => {
+  if that.isLeft then LeftX $ Z.concat (this.value, that.value) else that
+};
+let RightXprototypeXconcat = (this, that) => {
```
Comment (jceb) [inline index.serif:546]: Should we duplicate the above comment for consistency?

File: index.serif [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+#
+# # sanctuary-either
+#
+# The Either type represents values with two possibilities: a value of type
+# `Either a b` is either a Left whose value is of type `a` or a Right whose
+# value is of type `b`.
+
+# import {"$equals" as equals} from "../serif/src/prelude.js";
+import {} from "../serif/src/prelude.js";
+# import {Either as EitherY, Left as LeftY, Right as RightY, either as eitherY} from "../serif/src/prelude.js";
```
[the remark this answers is not in the document]
Comment (davidchambers) [reply index.serif:20]: The current algorithm is naive: it simply traverses the AST looking for identifiers. It doesn't differentiate between identifier introductions (1) and identifier references (2).

```javascript
    x => x
//  ^    ^
//  1    2
```

I can think of two possible workarounds:

```javascript
import {"Either" as EitherY, "Left" as LeftY, "Right" as RightY, "either" as eitherY} from "../serif/src/prelude.js";
```

```javascript
import prelude from "../serif/src/prelude.js";
```

File: test/index.js [hunk 4]
```diff
@@ -8,10 +8,11 @@ import Z            from 'sanctuary-type-classes';
 import type         from 'sanctuary-type-identifiers';
 import Useless      from 'sanctuary-useless';
 
-import Either       from '../index.js';
```
[the remark this answers is not in the document]
Comment (davidchambers) [reply test/index.js:11]: Yes, that seems like a good idea. :)

**Pull request closed without merging**